recipes/premake/5.x/conanfile.py

```diff
@@ -25,21 +24,23 @@ def source(self):
 
     @property
     def _platform(self):
-        return {'Windows': 'vs2017',
-                'Linux': 'gmake.unix',
-                'Macos': 'gmake.macosx'}.get(str(self.settings.os_build))
+        if self.version == "5.0.0-alpha14":
+            return {"Windows": "vs2017",
+                    "Linux": "gmake.unix",
+                    "Macos": "gmake.macosx"}.get(str(self.settings.os))
+        else:
+            return {"Windows": "vs2017",
+                    "Linux": "gmake2.unix",
+                    "Macos": "gmake2.macosx"}.get(str(self.settings.os))
 
     def build(self):
-        with tools.chdir(os.path.join(self._source_subfolder, 'build', self._platform)):
-            if self.settings.os_build == 'Windows':
+        with tools.chdir(os.path.join(self._source_subfolder, "build", self._platform)):
+            if self.settings.os == "Windows":
                 msbuild = MSBuild(self)
-                msbuild.build("Premake5.sln", platforms={'x86': 'Win32', 'x86_64': 'x64'}, build_type="Release", arch=self.settings.arch_build)
-            elif self.settings.os_build == 'Linux':
+                msbuild.build("Premake5.sln", platforms={"x86": "Win32", "x86_64": "x64"}, build_type="Release", arch="{}".format(self.settings.arch))
```

Comment:
* `arch` is not needed, it already takes the value from `conanfile.settings.arch`. 
* `build_type` (see note in main comment)

```suggestion
                msbuild.build("Premake5.sln", platforms={"x86": "Win32", "x86_64": "x64"})
```